decisions/2025-10-16_properties.md

```diff
@@ -0,0 +1,79 @@
+  d. options can be hierarchical, i.e current item categories in mSupply or organisation units in DHIS2 (to be confirmed)
+  e. tables which will have this property are specified, property can be re-used across many tables. (to be confirmed)
+
+2. User is able to set properties based on configurations, properties should all be clearable. 
+
+3. User can change property configurations, however migrations are too difficult to implement and maintain, it's well communicated that updates to configurations (like removing options or or changing option type) will not update existing value property values
+
+4. User will see properties in all of the views and lists/tables (as if it's another field/column that is part of a table)
+
+5. User will be able to filter by properties, in the natural option for the property type
+    a. Natural here suggests consistency with existing filters and aligning with expectation of how this property type should be filtered (for option type being able to filter by 'contains' might not be possible because the field might be translated, however would be possible if it's by general name)
+6. User can sort a table by a property set for that table
+7. User can configure and use properties at a particular store/site only (to be confirmed)
+
+There is also a requirement from customisation/plugin end to extend existing records for display/logic, we can feed two birds with ones scone having plugins and customisations also using properties, configuration/property setting and display UI for free. 
+
+## Options
+
+### Option 1
+
+As per below diagram.
+
+![media](media/properties.drawio.png)
+
+### Option 2
+
+Using JSON structures to store properties rather then a table. This would be harder to sort and filter and would require the whole record to be synced. 
+
+Also adding property configuration functionality to omSupply is slightly easier with "Option 1", however syncing properties attached to remote records is slightly harder (although properties for remote records would only be syncable with V7)
+
+### Option 3
+
+Have table specific property and/or property_value tables. The idea is to build for future performance in mind and reduce dynamic structured (usually safer and more readable/disoverable). However a pattern like would be very common and well understood, negating any effects this abstraction should have on readability/discoverability. As for performance, there is an assumption to be validated that table_name index together with record_id index will only speed up insert time (for index building), not query (query is the the performance concern, not insert), this goes in favour ot "Option 1".
+
+### Option 4
+
+Store value as JSON rather then different typed fields. This would be easier in general but if we need to do any calculations, it would be slower. Also if configuration is changed for the property it would require higher congnitive load and likely translations (to make sure existing values work)
```

Comment:
```suggestion
Store value as JSON rather than different typed fields. This would be easier in general but if we need to do any calculations, it would be slower. Also if configuration is changed for the property it would require higher cognitive load and likely translations (to make sure existing values work)
```